aws/terraform/modules/metaflow/modules/metadata-service/api-gateway.tf

```diff
@@ -0,0 +1,169 @@
+  }
+}
+
+resource "aws_api_gateway_method" "db" {
+  http_method      = "GET"
+  resource_id      = aws_api_gateway_resource.db.id
+  rest_api_id      = aws_api_gateway_rest_api.this.id
+  authorization    = "NONE"
+  api_key_required = var.api_basic_auth
+}
+
+resource "aws_api_gateway_integration_response" "this" {
+  rest_api_id = aws_api_gateway_rest_api.this.id
+  resource_id = aws_api_gateway_resource.this.id
+  http_method = aws_api_gateway_method.this.http_method
+  status_code = 200
+  depends_on  = [aws_api_gateway_integration.this]
+}
+
+resource "aws_api_gateway_integration" "this" {
+  rest_api_id = aws_api_gateway_rest_api.this.id
+  resource_id = aws_api_gateway_resource.this.id
+  http_method = aws_api_gateway_method.this.http_method
+
+  request_parameters = {
+    "integration.request.path.proxy" = "method.request.path.proxy"
+  }
+
+  type                    = "HTTP_PROXY"
+  uri                     = "http://${aws_lb.this.dns_name}/{proxy}"
+  integration_http_method = "ANY"
+  passthrough_behavior    = "WHEN_NO_MATCH"
+  connection_type         = "VPC_LINK"
+  connection_id           = aws_api_gateway_vpc_link.this.id
+}
+
+resource "aws_api_gateway_integration" "db" {
+  rest_api_id = aws_api_gateway_rest_api.this.id
+  resource_id = aws_api_gateway_resource.db.id
+  http_method = aws_api_gateway_method.db.http_method
+
+
+  type                    = "HTTP_PROXY"
+  uri                     = "http://${aws_lb.this.dns_name}:8082/db_schema_status"
+  integration_http_method = "GET"
+  passthrough_behavior    = "WHEN_NO_MATCH"
+  connection_type         = "VPC_LINK"
+  connection_id           = aws_api_gateway_vpc_link.this.id
+}
+
+resource "aws_api_gateway_method_response" "this" {
+  rest_api_id = aws_api_gateway_rest_api.this.id
+  resource_id = aws_api_gateway_resource.this.id
+  http_method = aws_api_gateway_method.this.http_method
+  status_code = "200"
+  depends_on  = [aws_api_gateway_integration.this]
+}
+
+resource "aws_api_gateway_method_response" "db" {
+  rest_api_id = aws_api_gateway_rest_api.this.id
+  resource_id = aws_api_gateway_resource.db.id
+  http_method = aws_api_gateway_method.db.http_method
+  status_code = "200"
+  depends_on  = [aws_api_gateway_integration.db]
+}
+
+resource "aws_api_gateway_deployment" "this" {
+  rest_api_id = aws_api_gateway_rest_api.this.id
+  stage_name  = local.api_gateway_stage_name
+  # explicit depends_on required to ensure module stands up on first `apply`
+  # otherwise a second followup `apply` would be required
+  # can read more here: https://stackoverflow.com/a/42783769
+  depends_on = [aws_api_gateway_method.this, aws_api_gateway_integration.this]
+
+  # ensures properly ordered re-deployments occur
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_api_gateway_api_key" "this" {
+  count = var.api_basic_auth ? 1 : 0
+  name  = local.api_gateway_key_name
+}
+
+resource "aws_api_gateway_usage_plan" "this" {
```

Comment:
It doesn't always happen, but there seems to be a race condition here.  Because there's no usage of the `aws_api_gateway_stage` resource in lieu of using `stage_name` in the deployment, there's nothing to link implicitly to the Usage Plan for `stage` besides the local variable.  The end result is that with only `aws_api_gateway_rest_api.this.id` as the implicit link, Terraform creates the usage plan with only the API itself as a dependency.  That means it sometimes creates the plan before the stage deployment is complete, causing a failure that only succeeds on a subsequent re-run.  I'd very much suggest tweaking this to use `aws_api_gateway_stage` in the flow, as it's considered a best practice by Terraform for a variety of reasons beyond dependency linking.  Quick reference to the docs here calling out the warning on using `stage_name`. https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment

Reply:
Resolving this in lieu of another quick fix.  Stay tuned!  :-)